pkg/stub/handler.go

```diff
@@ -606,23 +607,28 @@ func (h *Handler) CreateDefaultResourceIfNeeded(srcfg *v1alpha1.SamplesResource)
 	return srcfg, nil
 }
 
-func (h *Handler) WaitingForCredential(srcfg *v1alpha1.SamplesResource) error {
+func (h *Handler) WaitingForCredential(srcfg *v1alpha1.SamplesResource) (bool, error) {
 	if srcfg.ConditionTrue(v1alpha1.ImportCredentialsExist) {
-		return nil
+		return false, nil
 	}
 
 	// if trying to do rhel to the default registry.redhat.io registry requires the secret
 	// be in place since registry.redhat.io requires auth to pull; since it is not ready
 	// log error state
 	if srcfg.Spec.InstallType == v1alpha1.RHELSamplesDistribution &&
 		(srcfg.Spec.SamplesRegistry == "" || srcfg.Spec.SamplesRegistry == "registry.redhat.io") {
+		cred := srcfg.Condition(v1alpha1.ImageChangesInProgress)
+		if len(cred.Message) > 0 {
+			// need for credential already registered
+			return true, nil
+		}
```

Comment:
well the fact that there is a bug on line 620 does not help things :-) ... it should be looking at the message in the import credential condition, not the image in progress condition

In addition to fixing I'll expand on the comment along these lines:
- if import cred is false, and the message is emtpy, that means we have NOT registered the error, and need to do so
- if cred is false, and the message is there, we can just return nil to the sdk, to avoid additional churn;  once the secret comes in, it will update the samplesresource with cred == true, and then we'll get another samplesresource event that will trigger config processing